### PR TITLE
Search and update tasks

### PR DESCRIPTION
**Pull request by Paul-Yves**

### CODE REVIEW

File: lib/hubspot/task.rb [hunk 1]
```diff
@@ -33,6 +34,16 @@ def find(task_id, properties = DEFAULT_TASK_FIELDS)
...
         new(response)
       end
+
+      def search(properties = DEFAULT_TASK_FIELDS, body: {})
+        Hubspot::Connection.post_json(SEARCH_PATH, params: {}, body: { properties: }.merge(body))
+      end
+
+      def update!(task_id, properties = {})
+        request = { properties: properties }
+        response = Hubspot::Connection.patch_json(TASK_PATH, params: { task_id: }, body: request)
```
Comment (fonji) [inline lib/hubspot/task.rb:44]:
```suggestion
        request = { properties: }
        response = Hubspot::Connection.patch_json(TASK_PATH, params: { task_id: }, body: request)
```

or

```suggestion
        response = Hubspot::Connection.patch_json(TASK_PATH, params: { task_id: }, body: { properties: })
```

File: spec/lib/hubspot/task_spec.rb [hunk 2]
```diff
@@ -47,4 +47,39 @@
...
+    end
+  end
+
+  describe 'update!' do
+    let(:task_id) { 64_483_143_324 }
+    let(:properties) do
+      {
+        hs_task_status: 'COMPLETED'
+      }
+    end
```
Comment (fonji) [inline spec/lib/hubspot/task_spec.rb:74]:
```suggestion
    let(:properties) { { hs_task_status: 'COMPLETED' } }
```

File: spec/lib/hubspot/task_spec.rb [hunk 3]
```diff
@@ -47,4 +47,39 @@
...
+
+    subject(:update_task) { described_class.update!(task_id, properties) }
+
+    it 'updates existing task, returns the updated entity' do
+      VCR.use_cassette 'task_update' do
+        task = update_task
+        task.properties[:hs_task_status] = 'COMPLETED'
```
Comment (fonji) [inline spec/lib/hubspot/task_spec.rb:81]: this test needs an `expect`

**Review by fonji — COMMENTED** [review]
Cool!

**Review by fonji — APPROVED** [review]

**Review by PierreND — APPROVED** [review]

**Pull request merged**